src/js/server/lib/datasource.js

```diff
@@ -0,0 +1,161 @@
+gpii.express.user.couchdb.read.transformCouchResults = function (that, resp) {
+    // if undefined, pass that through as per dataSource (just for consistency in FS-backed tests)
+    var togo;
+    if (resp === undefined) {
+        togo = undefined;
+    } else {
+        if (resp.error) {
+            var error = {
+                isError: true,
+                message: resp.error + ": " + resp.reason
+            };
+            togo = fluid.promise();
+            togo.reject(error);
+        } else {
+            togo = fluid.model.transformWithRules(resp, that.options.rules.read);
+        }
+    }
+    return togo;
+};
+
+fluid.defaults("gpii.express.user.couchdb.read", {
+    gradeNames: ["kettle.dataSource.URL"],
+    writable:   false,
+    rules: {
+        read: {
+            "": ""
+        }
+    },
+    listeners: {
+        "onRead.transformCouchResults": {
+            funcName:  "gpii.express.user.couchdb.read.transformCouchResults",
+            args:      ["{that}", "{arguments}.0"],
+            namespace: "gpii.express.user.couchdb",
+            priority:  "after:encoding"
+        }
+    }
+});
+
+
+
+// A rewritten version of `kettle.dataSource.CouchDB.write` that decouples the read and write URLS (so that you can
+// read from a _view and write to a native Couch API.  The "read" URL is found at `options.urls.read`, and the "write" 
+// URL is set using `options.url.write`.   As this class  extends `kettle.dataSource.URL.writable`, `options.url` is 
+// set to the value of `options.url.write` by default.
+// 
+// Also allows rewriting of all payloads and responses using transformation rules.
+
+fluid.registerNamespace("gpii.express.user.couchdb.writable");
+
+// A replacement for the default couch "write" operation that removes the hard-coded `value` wrapper in favor of rules.
+gpii.express.user.couchdb.writable.transformModelToCouch = function (that, model, options) {
+    var jsonModel = typeof model === "string" ? JSON.parse(model) : model; // TODO: WHYYYYYYY?????
+    var doc = fluid.model.transformWithRules(jsonModel, that.options.rules.write);
+    var original = that.reader.get(that.reader.options.directModel, {filterNamespaces: ["JSON"]});
+    var togo = fluid.promise();
+    original.then(function (originalDoc) {
+        // We will always get a response even if the document is not found, so we have to check for more than just a value.
+        if (originalDoc && !originalDoc.error && originalDoc._id && originalDoc._rev) {
+            doc._id  = originalDoc._id;
+            doc._rev = originalDoc._rev;
+        } else {
+            options.writeMethod = "POST"; // returned out to URL dataSource handler
+        }
+        togo.resolve(JSON.stringify(doc));
+    });
+    return togo;
+};
+
+fluid.defaults("gpii.express.user.couchdb.writable", {
+    // We cannot directly extend the existing couch grade in Kettle because it has a non-namespaced listener for
```

Comment:
Yes, with the latest PR we can now use the Kettle writable couch grade, and I will do so in the "asymmetric" dataSource.  I will leave a TODO comment linking to that JIRA ticket.